src/psyclone/psyGen.py

```diff
@@ -3892,14 +4020,7 @@ def __init__(self, call, arg_info, access):
                 tag = "AlgArgs_" + self._text
 
                 # Prepare the Argument Interface Access value
-                if access and access.name == "READ":
-                    argument_access = ArgumentInterface.Access.READ
-                elif access and access.name == "WRITE":
-                    argument_access = ArgumentInterface.Access.WRITE
-                else:
-                    # If access is READWRITE, INC, SUM, UNKNOWN or no access
-                    # is given, use a READWRITE argument interface.
-                    argument_access = ArgumentInterface.Access.READWRITE
+                argument_access = ArgumentInterface.Access.READWRITE
```

Comment:
This was wrong. The Argument access.name gives the access attributes for inside the kernel being called while the Interface.Access gives the access attributes in the scope of the caller. And the interface can not be inferred from how it is used by a single kernel.

I think I introduced this mistake in a previous PR because the names are very similar. But then realized that the gen_code always considers algorithm arguments as read_write and this should do the same (otherwise creates invalid code).

Reply:
OK, thanks.